### PR TITLE
[createReactClass] remove createReactClass from RNTester/js/ImageExample.js

### PR DESCRIPTION
**Pull request by nd-02110114**
Related to #21581 .
Removed createReactClass from the RNTester/js/ImageExample.js

The diff of this PR is a little big. If there are any problems, please teach me 🙇 
## Test Plan
 - [x] npm run prettier
 - [x] npm run flow-check-ios
 - [x] npm run flow-check-android

## Release Notes
[GENERAL] [ENHANCEMENT] [RNTester/js/ImageExample.js] - remove createReactClass dependency

### CODE REVIEW

File: RNTester/js/ImageExample.js [hunk 1]
```diff
@@ -178,22 +206,38 @@ var NetworkImageExample = createReactClass({
...
-    Image.getSize(this.props.source.uri, (width, height) => {
-      this.setState({width, height});
-    });
+type ImageSizeExampleState = {
+  width: number,
+  height: number,
+};
+
+type ImageSizeExampleProps = $ReadOnly<{|
+  source: {
```
Comment (RSNara) [inline RNTester/js/ImageExample.js]:
```typescript
type ImageSizeExampleProps = $ReadOnly<{|
  source: $ReadOnly<{|
    uri: string,
  |}>
|}>
```

File: RNTester/js/ImageExample.js [hunk 2]
```diff
@@ -211,17 +255,54 @@ var ImageSizeExample = createReactClass({
...
+  }
+}
+
+type MultipleSourcesExampleState = {
+  width: number,
+  height: number,
+};
+
+class MultipleSourcesExample extends React.Component<
+  {},
```
Comment (RSNara) [inline RNTester/js/ImageExample.js]: nit: Can we move this out to `MultipleSourcesExampleProps`?

File: RNTester/js/ImageExample.js [hunk 3]
```diff
@@ -10,47 +10,67 @@
...
-  displayName: 'NetworkImageCallbackExample',
-  getInitialState: function() {
-    return {
-      events: [],
-      startLoadPrefetched: false,
-      mountTime: new Date(),
-    };
+
+type NetworkImageCallbackExampleState = {
+  events: string[],
```
[the remark this answers is not in the document]
Comment (nd-02110114) [reply RNTester/js/ImageExample.js]: 👍

File: RNTester/js/ImageExample.js [hunk 4]
```diff
@@ -10,47 +10,67 @@
...
+  mountTime: Date,
+};
+
+type NetworkImageCallbackExampleProps = $ReadOnly<{|
+  source: {
+    uri: string,
   },
+  prefetchedSource: {
+    uri: string,
+  },
```
[the remark this answers is not in the document]
Comment (nd-02110114) [reply RNTester/js/ImageExample.js]: Thank you!! 
I didn't understand `Flow's $ReadOnly (like Object.freeze) only goes 1 layer deep`

File: RNTester/js/ImageExample.js (hunk 1 above)
Comment (nd-02110114) [reply RNTester/js/ImageExample.js]: 👍

**Pull request closed without merging**